app/wall/page.tsx

```diff
@@ -132,40 +130,50 @@ export default function WallDisplay() {
     let updatedPVvalue: string | null | undefined = updatedPV.text;
 
     if (updatedPVName == instListPV && updatedPVbytes != null) {
-      // Act on an instlist change - subscribe to each instrument's runstate PV.
-      const dehexedInstList = dehex_and_decompress(atob(updatedPVbytes));
-      if (dehexedInstList != null && typeof dehexedInstList == "string") {
-        const instListDict = JSON.parse(dehexedInstList);
-        for (const item of instListDict) {
-          // Iterate through the instlist, find their associated object in the ts1data, ts2data or miscData arrays, get the runstate PV and subscribe
-          const instName = item["name"];
-          const instPrefix = item["pvPrefix"];
-          const foundInstrument = [...TS1Data, ...TS2Data, ...miscData].find(
-            (instrument) => instrument.name === instName,
-          );
-          if (foundInstrument) {
-            // Subscribe to the instrument's runstate PV
-            foundInstrument.pv = instPrefix + runstatePV;
-            sendJsonMessage({ type: "subscribe", pvs: [foundInstrument.pv] });
-          }
-        }
-      }
-    } else {
-      // An instrument's runstate has changed. Find the instrument and update its status (if there is one!).
-      const foundInstrument = [...TS1Data, ...TS2Data, ...miscData].find(
-        (instrument) => instrument.pv === updatedPVName,
-      );
-      if (updatedPVvalue && foundInstrument) {
-        foundInstrument.status = updatedPVvalue;
+      const instListDict = instListFromBytes(updatedPVbytes);
+      for (const item of instListDict) {
+        // Iterate through instruments in the instlist, get the runstate PV and subscribe
+        const instName = item["name"];
+        const instPrefix = item["pvPrefix"];
+        setData((prev) => {
+          const newData: Array<targetStation> = [...prev];
+          newData.map((targetStation) => {
+            const foundInstrument = targetStation.instruments.find(
+              (instrument) => instrument.name === instName,
+            );
+            if (foundInstrument) {
+              foundInstrument.runstatePV = instPrefix + runstatePV;
+              // Subscribe to the instrument's runstate PV
+              sendJsonMessage({
+                type: PVWSRequestType.subscribe,
+                pvs: [foundInstrument.runstatePV],
+              });
+            }
+          });
```

Comment:
Are you sure this is a correct usage of `.map`?

We're not binding anything to the result of this call, if that's correct and we really want to be mutating the original then `forEach` maybe clearer. `.map` is usually recommended to explicitly avoid mutating the originals though?

Reply:
yeah a forEach is much clearer and we've already copied with the spread above (`[...prev]`) 
will implement.

Reply:
Thanks, there were a couple of other places like this, only commented on one